packages/meditrak-server/src/database/SurveyResponseChangeHandler/SurveyResponseChangeHandler.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { OutdatedStatusUpdater } from './OutdatedStatusUpdater';
+
+const UPDATE_DEBOUNCE_TIME = 250; // wait 250ms after changes before updating, to avoid double-up
+
+export class SurveyResponseChangeHandler {
```

Comment:
I'd like to get some feedback on this one: while the name of the class is generic, it currently only handles changes specifically related to keeping the `outdated` status up to date. This could change in the future though, by adding more handlers.

There is one more change handler that is currently defined in the [SurveyResponse.js](https://github.com/beyondessential/tupaia/blob/dev/packages/meditrak-server/src/database/models/SurveyResponse.js#L101) model but could (should?) be moved here. I've found the fact that notification handlers may be defined in multiple places confusing in the past.

Do you reckon this is fine as is or should we:
* a) move the notification from the model here, or
* b) leave the notification as is and rename this class to `SurveyResponseOutdatedStatusUpdater` or similar. I started with this approach but then changed it because I didn't like the class name, and I've found that ugly names may be an indication of wrong abstractions

Reply:
I'd lean towards b) however I'd probably also wanna move that change hander from `SurveyResponse` to another class beside here called something like `AnswerChangePublisher`?

As for the class name being ugly, how does `SurveyResponseOutdater` sound? It kinda fits with our existing naming convention of `<noun (often model)><verb>`, ie. `AnalyticsRefresher`, `EntityHierarchyCacher`?

I'm with you that having change notification handlers scattered around the place can be confusing. And I think over time they may grow more similar functionally (sharing debounce logic code and such) for performance reasons.

Reply:
While it might seem a bit odd, I think the convention for these 'change handler routine' classes is for them to be defined in the `@tupaia/database` package (see. `AnalyticsRefresher`, `EntityHierarchyCacher`). Even if they are instantiated by the meditrak-server, I think they are meant to be agnostic towards that, and are solely a 'database' concern? 

Happy to get pushback on that though since I do think the pattern is a bit un-intuitive

Reply:
A few points here:
1. Love the name `SurveyResponseOutdater`, great suggestion @rohan-bes 
2. Agree that it's getting confusing having notification responders in lots of different places. This is starting to become more significant because a) they're a source of performance bottleneck if used naively, and b) when we scale horizontally, we'll need to make sure only one of the instances is performing the side effects, so having many places to turn that switch on isn't great. We can review where change handlers sit more holistically when looking at horizontal scaling.
3. I'm keen to keep the change handler routines in `@tupaia/database` as you suggest @rohan-bes, mainly because we may decide to have a completely separate "change handler" service that responds to notifications and makes side effects (because of the horizontal scaling point above), plus I don't think this is necessarily a `meditrak-server` responsibility anyway (it probably _is_ the most natural fit, but mainly because it's a catch-all)

Reply:
I'm thinking along the same lines with you on 3. @edmofro and another concern of mine is that we may need to evolve to a slightly more advanced messaging protocol soon enough, and if so having all the subscribers/publishers in the same place alongside the event definitions might be helpful (think redux actions/reducers).

I say this because I'm a bit worried that we may start seeing some nasty feedback loops between this handler and the `AnalyticsRefresher`, and I think such issues may just become more common over time. 

Ofcourse let's see how it goes in testing with large data submissions before we get too worried, just set Juliana on it for a day 😝